components/service/firefox-accounts/src/test/java/mozilla/components/service/fxa/FxaAccountManagerTest.kt

```diff
@@ -1886,6 +1887,180 @@ class FxaAccountManagerTest {
         verify(syncManager, times(2)).stop()
         verifyNoMoreInteractions(syncManager)
     }
+    @Test
+    fun `beginAuthenticationAuthSync handles any pairingUrl correctly`() {
+        // Initialize things so the test can run
+        WorkManagerTestInitHelper.initializeTestWorkManager(testContext)
+
+        // Build our dependencies for creating an account manager
+        val tServerConfig = ServerConfig(
+                "https://localhost/contentUrl",
+                "testClientId",
+                "https://localhost/redirectUrl"
+        )
+        val tDeviceConfig = DeviceConfig(
+                "testDevice",
+                DeviceType.MOBILE,
+                setOf()
+        )
+        val tSyncConfig = SyncConfig(setOf(SyncEngine.Bookmarks))
+        val manager = FxaAccountManager(
+                testContext,
+                serverConfig = tServerConfig,
+                deviceConfig = tDeviceConfig,
+                syncConfig = tSyncConfig
+        )
+        val pairingUrl = "https://localhost/pairingUrl".toString()
+        val beginResponse = manager.beginAuthenticationAsync(pairingUrl)
+        assertTrue(beginResponse is CompletableDeferred<String?>)
```

Comment:
Unless you `await` for this to resolve, you're not really testing anything. This `CompletableDeferred<String?>` is just a promise to return a value of `String?` at some point in the future, e.g. once this API request goes through the internal state machine.

What you want to do is wrap this whole method in `runBlocking`, set the corresponding coroutine context on your test account manager (`FxaAccountManager(... coroutineContext = this@runBlocking.coroutineContext ...)` and do something like `assertEquals("auth://url", manager.beginAuthenticationAsync().await())`. This will ensure that the internal account manager state machine will actually go through its motions in response to  a beginAuthenticationAsync call, using the same dispatcher as is used for your test code.

However, lots of other tests in this class already exercise `beginAuthenticationAsync`, so as-is this test isn't adding any value.

Name of the test says "any pairing url", however you're just passing in one. Did you intend to test `beginAuthenticationAsync` against a few different inputs?